syncto/authentication.py

```diff
@@ -48,8 +48,20 @@ def build_sync_client(request):
     if not credentials:
         ttl = int(settings['syncto.cache_credentials_ttl_seconds'])
         tokenserver = TokenserverClient(bid_assertion, client_state)
+        if request.registry.statsd:
+            request.registry.statsd.watch_execution_time(tokenserver,
+                                                         prefix="tokenserver")
         credentials = tokenserver.get_hawk_credentials(duration=ttl)
         request.registry.cache.set(cache_key, credentials, ttl)
 
+    if request.registry.statsd:
```

Comment:
You can also add a default setting for statsd to avoid testing if it's setup, no ?